### ISSUE TITLE
HarmonySpan not triggering as expected

### ISSUE THREAD

**Issue by nbr0T** [AShifter/harmony-span#28]
Had a smooth installation and HarmonySpan detects my Harmony Elite keypresses as well as keypresses from the Harmony mobile app (Android, Samsung S9+), however no trigger seems to be send from the HarmonySpan although it refers to the expected Post/event.

A simple setup with the event “lighton” is configured as:
Webhook name: lighton
Webhook URL: https://maker.ifttt.com/use/_mypersonalkey_
The URL is copied from the webhooks setting on the ifttt.com.
HarmonySpan is granted the appropriate network access and even running in administrator mode.

I tried triggering the event ”lighton” manually from the PC (win10) running HarmonySpan with the cmd:
“curl -X POST https://maker.ifttt.com/trigger/lighton/with/key/_mypersonalkey_”
That worked fine and turned the light on, thus I came to wonder whether I am using the Webhook URL in HarmonySpan correctly?
Any logs to be collected that might by helpful?

**Comment by kertofer** [comment]
I am having this same issue. It appears as though the webhooks are not being sent by the app after the app acknowledges the receipt of the button press from my harmony remote. As nbr0T said above if I use curl everything works.

**Comment by snarknesor** [comment]
same here. did you ever find a solution?

**Comment by kertofer** [comment]
> same here. did you ever find a solution?

Yes, I pulled the latest commit, compiled it and ran it manually instead of using the pre compiled binary. The interface is a bit cleaner and it runs perfectly for me now.

**Comment by michaelewers** [comment]
There are not binaries for the latest version. AShifter wanted to provide them some months ago.

To run the latest version "manually" follow these steps:

- Install Node.js on your platform
- Install git on your platform (the git client)
- open a console window
- type "git clone https://github.com/AShifter/harmony-span.git" to download the source code to your machine
- type "cd harmony-span" to enter the directory where the downloaded source code is
- type "npm install" to install the dependencies that the program needs
- type "npm start" to start the program
Then follow the instructions on the screen.
Use Ctrl-C to stop the program.

From there you can always use "git pull" in the harmony-span directory to update to the latest version.

Note that the config file format has changed from json to yaml, so your old config.json will no longer work.

With the latest version I have fixes some stuff so chances are that your issue will be resolved.

**Comment by snarknesor** [comment]
that worked. Thanks!

**Comment by Trevorwelton** [comment]
Is there going to be a binary update at some point?

I had this working for a bit now when I click the pause button on my remote I can see that it is triggering the sequence but the lights don't come on.  Also, I could never get lights to come on with the play button despite seeing the sequence triggered and the webhooks test working.  

Seems pretty buggy?  Will the changes listed above help my situation?

**Comment by amathews84** [comment]
Hello, not sure if this is a related but I see the triggers firing and the expected http post event executing (somewhat reliably) when the Windows harmony span terminal is open but the post event does not execute when the terminal is not open on my computer.  

Is this related to the required binary update?
